src/solvers/refinement/string_constraint.cpp

```diff
@@ -0,0 +1,21 @@
+/*******************************************************************\
+
+ Module: String solver
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include "string_constraint.h"
+
+string_constraintt::string_constraintt(
+  symbol_exprt _univ_var,
+  exprt lower_bound,
+  exprt upper_bound,
+  exprt body)
```

Comment:
I think all of the above should be passed as const references; arguably that's not something you've newly introduced, but please fix it.

Reply:
const reference may not be a good idea because the constraint could escape the scope in which these expressions are defined, a better solution may be to use `move` below and in places where the constraint are constructed.

Reply:
Sorry, the original suggestion makes sense because you have a copy below, so that would avoid one copy. I'd still prefer to avoid const reference (and use move) so that arguments don't need to be lvalues.

Reply:
Changed to use const references.